tensorflow/stream_executor/platform/default/dso_loader.cc

```diff
@@ -41,6 +45,38 @@ string GetDirectMLPath() {
   return (path != nullptr ? path : "");
 }
 
+#if _WIN32
+string GetModuleDirectory() {
+  HMODULE tensorflowHmodule = nullptr;
+  BOOL getHandleResult = GetModuleHandleExW(
+      GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT |
+          GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS,
+      reinterpret_cast<LPCWSTR>(&GetModuleDirectory), &tensorflowHmodule);
+  CHECK_EQ(getHandleResult, TRUE);
+
+  // Safe const_cast because of explicit bounds checking and contiguous memory
+  // in C++11 and later.
+  std::wstring wpath(MAX_PATH, '\0');
+  DWORD filePathSize =
+      GetModuleFileNameW(tensorflowHmodule, const_cast<wchar_t*>(wpath.data()),
+                         static_cast<DWORD>(wpath.size()));
+  if (filePathSize > wpath.size()) {
```

Comment:
>filePathSize > wpath.size() [](start = 6, length = 27)

This should check if GetLastError() == ERROR_INSUFFICIENT_BUFFER and use a loop, doubling the buffer each iteration. This is because if the buffer is too small, GetModuleFileName(h, p, len) returns len.

Reply:
Nice catch! You found _two_ bugs, since this isn't original code. :)